docs(keyring) add recovery key for 3.0 (#4053)
Kong/docs.konghq.com: src/gateway/plan-and-deploy/security/db-encryption.md

purposes. It is not designed to be modified or decrypted before being used during
a disaster recovery process.

### Restore the Keyring

As the Keyring material is encrypted with a randomly-generated symmetric key, and only stored in memory,
the Keyring could be lost or corrupted if the Kong process is restarted.
Either way, the Keyring can be restored by importing the exported keyring.

> A recommended procedure is to use the Recovery mode, you should set the `keyring_recovery_public_key`

Reviewer: ```suggestion
A recommended method for recovering a keyring is to use the Recovery mode, you should set the `keyring_recovery_public_key`
```

> Kong configuration value to the public key of the recovery keypair.

Reviewer: ```suggestion
Kong configuration value to the public key of the recovery key pair.
```

> (You can also follow the Generate a Management RSA Key Pair guide to generate a keypair for this purpose.)

Reviewer: Can we link the doc here?

> The Keyring material is encrypted with the public RSA key defined via the `keyring_recovery_public_key`

Reviewer: ```suggestion
The keyring material is then encrypted with the public RSA key defined via the `keyring_recovery_public_key`
```

> Kong configuration value in the database, the recovery keypair then is used to decrypt the Keyring material in the database.

Reviewer: ```suggestion
Kong configuration value in the database. You can use this recovery key to decrypt the Keyring material in the database.
```
